### PR TITLE
[Merged by Bors] - Update discv5

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed

Updates discv5

Pending on
- [x] #3547 
- [x] Alex upgrades his deps

## Proposed Changes

updates discv5 and the enr crate. The only relevant change would be some clear indications of ipv4 usage in lighthouse

## Additional Info

Functionally, this should be equivalent to the prev version.
As draft pending a discv5 release

### CODE REVIEW

**Comment by AgeManning** [comment]
Bumping this. @sean - it looks like Alex's repo is causing a bunch of pain for us to maintain. Is it correct that we are only using his deps for testing?

**Comment by michaelsproul** [comment]
@AgeManning Could we hack in our forks of the deps using a top-level `[patch]` in `Cargo.toml`?

(btw you tagged some rando Sean, our Sean is @realbigsean :rofl: )

**Comment by divagant-martian** [comment]
@AgeManning @michaelsproul alex helped me upgrading his deps and this should be ready now

File: beacon_node/lighthouse_network/src/discovery/enr_ext.rs [hunk 1]
```diff
@@ -232,6 +232,9 @@ impl CombinedKeyExt for CombinedKey {
                         .expect("libp2p key must be valid");
                 Ok(CombinedKey::from(ed_keypair))
             }
+            Keypair::Ecdsa(_) => {
+                todo!("What here??")
```
Comment (divagant-martian) [inline beacon_node/lighthouse_network/src/discovery/enr_ext.rs]: @AgeManning I didn't remember until I did an extra review, but it seems this needs your input

File: beacon_node/lighthouse_network/src/discovery/enr_ext.rs [hunk 2]
```diff
@@ -265,6 +268,9 @@ pub fn peer_id_to_node_id(peer_id: &PeerId) -> Result<discv5::enr::NodeId, Strin
             hasher.finalize(&mut output);
             Ok(discv5::enr::NodeId::parse(&output).expect("Must be correct length"))
         }
+        PublicKey::Ecdsa(_) => {
+            todo!("what here?")
```
Comment (divagant-martian) [inline beacon_node/lighthouse_network/src/discovery/enr_ext.rs]: @AgeManning same here

**Comment by divagant-martian** [comment]
@AgeManning rebased and squashed so that the diff makes sense now, but I see I have two TODOs that need your input

File: beacon_node/lighthouse_network/src/discovery/enr_ext.rs (hunk 1 above)
Comment (AgeManning) [reply beacon_node/lighthouse_network/src/discovery/enr_ext.rs]: Looks like they've added a new keypair type. We haven't built the machinery to handle this, so just throw an error saying "invalid key type"

File: beacon_node/lighthouse_network/src/discovery/enr_ext.rs (hunk 2 above)
Comment (AgeManning) [reply beacon_node/lighthouse_network/src/discovery/enr_ext.rs]: Yeah same here, just throw an error, "invalid key type"

**Comment by AgeManning** [comment]
@divagant-martian - Think we just throw errors. IIRC we handle the error cases fairly gracefully.

**Comment by divagant-martian** [comment]
friendly ping @AgeManning

**Review by AgeManning — APPROVED** [review]
This looks good to me. Ready to merge, when release schedule allows

**Comment by michaelsproul** [comment]
bors retry

**Pull request closed without merging**